content/blog/patchwork-1-3-0/index.Rmd

```diff
@@ -0,0 +1,235 @@
+  theme(axis.text.x = element_blank(), axis.title.x = element_blank()) +
+  scale_x_discrete(expand = c(0, 0.5))
+
+# Construct our table
+table <- rbind(
+  tapply(airquality$Temp, airquality$Month, max),
+  tapply(airquality$Temp, airquality$Month, median),
+  tapply(airquality$Temp, airquality$Month, min)
+)
+colnames(table) <- month.name[5:9]
+table <- data.frame(
+  Measure = c("Max", "Median", "Min"),
+  table
+)
+table <- gt(table, rowname_col = "Measure") |>
+  cols_width(contains(month.name) ~ px(100)) |>
+  cols_align(align = "center") |>
+  cols_align(align = "right", columns = "Measure")
+
+p2 / wrap_table(table, space = "fixed")
+```
+
+Circling back, there was another argument to `wrap_table()` we didn't get into yet. In the plot above, we see that the row names are conveniently aligned with the axis rather than the panel of the plot above, in the same way as the headers where placed outside the panel area. This is a nice default and generally makes sense for the semantics of a table, but you might want something different. The `panel` argument allows you to control this exact behavior. It takes `"body"`, `"full"`, `"rows"`, or `"cols"` which indicate what portion of the table should be inside the panel area. The default is `"body"` which places row and column names outside the panel. `"full"`, on the contrary, places everything inside, while `"rows"` and `"cols"` are half versions that allows you to keep either column *or* row names outside the panel respectively.
+
+```{r}
+# Place all rows (including the header row) inside the panel area
+p1 + wrap_table(aq, panel = "rows", space = "free_y")
+```
+
+Just like the tables support ggplot2-like titles, they also support tags, meaning that patchworks auto-tagging works as expected. It can be turned off using the `ignore_tag` argument but often you'd want to treat it as a figure in the figure text:
+
+```{r}
+p1 + wrap_table(aq, panel = "rows", space = "free_y") +
+  plot_annotation(tag_levels = "A") &
+  theme(plot.tag = element_text(margin = margin(0, 6, 6, 0)))
+```
+
+### Future
+
+The support on the patchwork end is likely done at this point, but the conversion to grobs that has been added to gt is still somewhat young and will improve over time. It is likely that markdown formatting (through marquee) and other niceties will get added, leading to even more power in composing tables with plots using patchwork as the glue between them.
+
+## With freedom comes great responsibility
+
+The second leg of this release concerns the `free()` function which was introduced in the last release. I devoted a whole section of my posit::conf talk this year to talk about `free()` and how it was a good thing to say no to requests for functionality until you have a solution that fits into your API and doesn't add clutter. I really like how the API for `free()` turned out but I also knew it could do more. In this release it delivers on those promises with two additional arguments being added to it.
```

Comment:
```suggestion
The second leg of this release concerns the `free()` function which was introduced in the last release. I devoted a whole section of my posit::conf talk this year to talk about `free()` and how it was a good thing to say no to requests for functionality until you have a solution that fits into your API and doesn't add clutter. I really like how the API for `free()` turned out but I also knew it could do more. In this release it delivers on those promises with two additional arguments.
```